pkg/storage/stores/shipper/compactor/retention/retention.go

```diff
@@ -341,16 +349,16 @@ func (c *chunkRewriter) rewriteChunk(ctx context.Context, ce ChunkEntry, interva
 		newChunk := chunk.NewChunk(
 			userID, chks[0].FingerprintModel(), chks[0].Metric,
 			facade,
-			interval.Start,
-			interval.End,
+			start,
+			end,
 		)
 
 		err = newChunk.Encode()
 		if err != nil {
 			return false, err
 		}
 
-		entries, err := c.seriesStoreSchema.GetChunkWriteEntries(interval.Start, interval.End, userID, "logs", newChunk.Metric, c.scfg.ExternalKey(newChunk.ChunkRef))
+		entries, err := c.seriesStoreSchema.GetChunkWriteEntries(start, end, userID, "logs", newChunk.Metric, c.scfg.ExternalKey(newChunk.ChunkRef))
```

Comment:
I think we should be using the intervals from the new chunk and not the rebound interval since with line filters we might have new chunk bounds.

Reply:
Done.